vumi_http_proxy/test/test_http_proxy.py

```diff
@@ -36,59 +34,23 @@ def make_request(self, proxy_port, url):
         returnValue((response, body))
 
     @inlineCallbacks
-    def check_proxy_request(self, blacklist, expected_code, expected_body):
+    def check_proxy_request(self, blacklist, ip, expected_code, expected_body):
         http_port = yield self.server.start()
         proxy_port = yield self.setup_proxy(blacklist)
-        url = 'http://127.0.0.1:%s/' % (http_port,)
+        url = 'http://%s:%s/' % (ip, http_port,)
         response, body = yield self.make_request(proxy_port, url)
         self.assertEqual(response.code, expected_code)
         self.assertEqual(body, expected_body)
 
-    def test_allow(self):
-        return self.check_proxy_request([], 200, '<html>Allowed</html>')
-
-    def test_deny(self):
+    def test_badIP(self):
```

Comment:
We should probably change the name of this test to `test_bad_ip` so that it matches our conventions (we follow PEP8 -- even though Twisted doesn't).